src/parlant/adapters/vector_db/transient.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2024 Emcie Co Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+import asyncio
+import json
+from typing import Any, Generic, Mapping, Optional, Sequence, cast
+import numpy as np
+from typing_extensions import override, Self
+
+import nano_vectordb  # type: ignore
+
+from parlant.core.nlp.embedding import Embedder, EmbedderFactory
+from parlant.core.logging import Logger
+from parlant.core.persistence.common import ensure_is_total, matches_filters, Where
+from parlant.core.persistence.vector_database import (
+    BaseDocument,
+    DeleteResult,
+    InsertResult,
+    SimilarDocumentResult,
+    UpdateResult,
+    VectorCollection,
+    VectorDatabase,
+    TDocument,
+)
+
+
+class TransientVectorDatabase(VectorDatabase):
+    def __init__(
+        self,
+        logger: Logger,
+        embedder_factory: EmbedderFactory,
+        embedder_type: type[Embedder],
+    ) -> None:
+        self._logger = logger
+        self._embedder_factory = embedder_factory
+        self._embedder_type = embedder_type
+
+        self._multi_tenant: nano_vectordb.MultiTenantNanoVDB
+        self._collection_name_to_tenant_id: dict[str, str] = {}
+        self._collections: dict[str, _TransientCollection[BaseDocument]] = {}
+
+    async def __aenter__(self) -> Self:
+        self._embedder = self._embedder_factory.create_embedder(self._embedder_type)
+        self._multi_tenant = nano_vectordb.MultiTenantNanoVDB(self._embedder.dimensions)
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[object],
+    ) -> None:
+        pass
+
+    @override
+    async def create_collection(
```

Comment:
You're returning a private class. It's a design bug to expose a private class. Either return `VectorCollection` or make `TransientVectorCollection` public (no underscore prefix).